[IMP] Manufacturing: Kit article updates (#12162)
odoo/documentation: content/applications/inventory_and_mrp/manufacturing/advanced_configuration/kit_shipping.rst

> Create the kit as a product

Reviewer: I can't comment on the line, but let's rename this article to simply "Kits". This was named when the naming convention was "[Verb] [Feature]", and it's now simply "[Feature]"

Author: Updated

===========================

To use a kit as a sellable product, or as a component organization tool, the kit should first
be created as a product.

> To create a kit product, go to :menuselection:`Inventory app --> Products --> Products`, then select

Reviewer: The product form can also be accessed from the Manufacturing app, and since this doc is located in the MRP section, let's just go there instead (:

```suggestion
To create a kit product, go to :menuselection:`Manufacturing app --> Products --> Products`, then select
```

Create the kit as a product
===========================

To use a kit as a sellable product, or as a component organization tool, the kit should first be
created as a product.

To create a kit product, go to :menuselection:`Manufacturing app --> Products --> Products`, then
select :guilabel:`New`.

Enter a name to the new kit product. Next, under the :guilabel:`General Information` tab, set the
:guilabel:`Product Type` to :guilabel:`Goods`. Then, select the checkbox next to :guilabel:`Track

> Inventory`. From the drop-down menu, select :guilabel:`By Quantity`.

Author: I'll implement, but I agree that this is something to look at whenever we do an instruction path in paragraph form instead of list form. I don't have a good reason for why it should be shorter vs longer sentences besides separating the actions out a little bit more.

==================

After fully configuring the kit product and its components, a new |BOM| can be created for the kit
product.

To do so, go to :menuselection:`Manufacturing app --> Products --> Bills of Materials`, then select
:guilabel:`New`. Next to the :guilabel:`Product` field, select the drop-down menu to reveal a list
of products, then select the previously configured kit product.

Then, for the :guilabel:`BoM Type` field, select the :guilabel:`Kit` option. Finally, under the
:guilabel:`Components` tab, select :guilabel:`Add a line`, then add each desired component and
specify their quantities under the :guilabel:`Quantity` column.

Once ready, select :guilabel:`Save` to save the newly created |BOM|. Learn more about :doc:`bills of
materials <../basic_setup/bill_configuration>`.

.. image:: kit_shipping/bom-kit-selection.png
   :alt: Kit selection on the bill of materials.

If the kit is solely being used as a sellable product, then only components need to be added under
the :guilabel:`Components` tab, and configuring manufacturing operations is not necessary. Kits can
be added as components to other kits, which is described in the next section.

.. note::

>    When a kit is sold as a product, it appears as a single line item on the quotation and sales

Reviewer: The paragraph above the note ("If the kit is solely being used as a sellable product, then only components need to be added...") is  in between an image and note block, it's a little hard to read. 

Let's move all the text above the image to improve readability! While we typically keep admonition blocks above images, feel free to use your judgement when rewriting and reordering the info in this section (:

After fully configuring the kit product and its components, a new :abbr:`BoM (bill of materials)`
can be created for the kit product.

To do so, go to :menuselection:`Manufacturing app --> Products --> Bills of Materials`, then select
:guilabel:`New`. Next to the :guilabel:`Product` field, select the drop-down menu to reveal a list
of products, then select the previously configured kit product.

Then, for the :guilabel:`BoM Type` field, select the :guilabel:`Kit` option. Finally, under the
:guilabel:`Components` tab, select :guilabel:`Add a line`, then add each desired component and
specify their quantities under the :guilabel:`Quantity` column.

> Once ready, select :guilabel:`Save` to save the newly-created :abbr:`BoM (bill of materials)`.

Reviewer: Underneath this, possibly add a seealso admonition block linking to the BoM configuration article

Absolute local path: `content/applications/inventory_and_mrp/manufacturing/basic_setup/bill_configuration.rst`

Author: Added inline since it looked a little weird with the image right below it and the next note coming so soon after. lmk if you have strong feelings about this being a callout though!

Kits are also used to manage multi-level :abbr:`BoMs (bills of materials)`. These are products that

> contain **other** :abbr:`BoM (bill of materials)` products as components, and therefore require

Reviewer: I just had a realization: I heard from partner teams that customers are looking for ways to configure multi-level BoMs and I didn't realize that we were missing that doc until reading your PR! 

Creating another task for you (:

Author: :3

> *nested* :abbr:`BoMs (bills of materials)`. Incorporating preconfigured kits into multi-level

Reviewer: This is just a helpful tip, rather than an issue, but do you ever get tired of typing the abbreviation tag and contents over and over?

We can use a short hand! So at the beginning & top of the file, right below the heading, we can define some replace terms

```
.. |BOM| replace:: :abbr:`BoM (Bill of Materials)`
.. |BOMs| replace:: :abbr:`BoMs (Bills of Materials)`
.. |MO| replace:: :abbr:`MO (Manufacturing Order)`
```

Then, in the content, we can just do "... *nested* |BOMS|. Incorporating preconfigured kits into multi-level..."

![image](https://github.com/user-attachments/assets/0f5c28bd-0630-4c92-92f4-fcd17ecd4f48)

Author: Added those in :)

:abbr:`BoMs (bills of materials)` allows for cleaner organization of bundled products.

To configure this type of :abbr:`BoM (bill of materials)` with a kit as a component, go to
:menuselection:`Manufacturing app --> Products --> Bills of Materials`, then select
:guilabel:`New`.

Next to the :guilabel:`Product` field, select the drop-down menu to reveal a list of products, and
select the desired :abbr:`BoM (bill of materials)` product. Then, for the :guilabel:`BoM Type`
field, select the :guilabel:`Manufacture this product` option.

Under the :guilabel:`Components` tab, select :guilabel:`Add a line`, then select a kit as the
component. Adding the kit as a component eliminates the need to add the kit's components
individually. Any :guilabel:`BoM Type` can be used for the higher-level product's :abbr:`BoM (bill
of materials)`.

Once ready, select :guilabel:`Save` to save changes.

.. image:: kit_shipping/multilevel-bom-kit.png
   :alt: Kit as a component in a multilevel bill of materials.

Preview multi-level BoMs
------------------------

To access a comprehensive overview of the multi-level :abbr:`BoM's (bill of material's)` components,

> select on the :guilabel:`BoM Overview` smart button. Sublevel :abbr:`BoMs (bills of materials)`

Reviewer: Just a quick reminder for readers on where to find the BoM Overview smart button if they weren’t already on the BoM page. We usually provide these reminders (when it makes sense) since some readers tend to skim or jump ahead. 😊

```suggestion
go to the intended :abbr:`BoM (bill of materials)` and select the :guilabel:`BoM Overview` smart button. Sublevel :abbr:`BoMs (bills of materials)`
```

Author: Updated

can be expanded and viewed from this report.

.. image:: kit_shipping/bom-overview-kit.png
   :alt: Expanded kit in the Bill of Materials overview.

When creating a manufacturing order for a product with a multi-level :abbr:`BoM (bill of
materials)`, the kit product automatically expands to show all components. Any operations in the
kit's :abbr:`BoM (bill of materials)` are also added to the list of work orders on the
manufacturing order.

.. tip::

>    Kits are primarily used to bundle components together for organization or sale. However, you can use :doc:`sub-assemblies<sub_assemblies>` to manage

Reviewer: We wanna have a space between link <> and link text (:
```suggestion
   Kits are primarily used to bundle components together for organization or sale. However, you can use :doc:`sub-assemblies <sub_assemblies>` to manage
```

Author: Ah ty for this callout. I've been [using this website to learn rST](https://sublime-and-sphinx-guide.readthedocs.io/en/latest/references.html), but I see [the example here](https://sublime-and-sphinx-guide.readthedocs.io/en/latest/references.html) has that space. I'll need to go update the other PRs!
